internal/rbd/replicationcontrollerserver.go

```diff
@@ -104,6 +119,43 @@ func getMirroringMode(ctx context.Context, parameters map[string]string) (librbd
 	return mirroringMode, nil
 }
 
+// getSchedulingDetails gets the mirroring mode and scheduling details from the
+// input GRPC request parameters and validates the scheduling is only supported
+// for mirroring mode.
+func getSchedulingDetails(parameters map[string]string) (admin.Interval, admin.StartTime, error) {
+	val := parameters[imageMirroringKey]
+
+	switch imageMirroringMode(val) {
+	case imageMirrorModeSnapshot:
+	default:
+		return "", "", status.Error(codes.InvalidArgument, "scheduling is only supported for snapshot mode")
+	}
+
+	startTime := parameters[schedulingStartTimeKey]
+	// validate mandatory interval field
+	interval, ok := parameters[schedulingIntervalKey]
+	if ok && interval == "" {
+		return "", "", status.Error(codes.InvalidArgument, "scheduling interval cannot be empty")
+	}
+	if !ok {
+		// startTime is alone not supported it has to be present with interval
+		if startTime != "" {
+			return "", "", status.Errorf(codes.InvalidArgument, "%s parameter is supported only with %s", schedulingStartTimeKey, schedulingIntervalKey)
+		}
+	}
+	return admin.Interval(interval), admin.StartTime(startTime), nil
+}
+
+// formatSchedulingInterval return the interval as it is if its ending with
+// `m|h|d` or else it will add `m` as default because ceph adds `m` by default.
+func formatSchedulingInterval(interval string) admin.Interval {
+	var re = regexp.MustCompile(`m|h|d`)
```

Comment:
do you want to match a `<digit><m|h|d><EOL>` here? You might be able to write it as `^[0-9]+[mhd]$` in that case and be more strict.